src/compat-shallow-renderer/PreactShallowRenderer.ts

```diff
@@ -168,6 +166,7 @@ export default class PreactShallowRenderer {
   private _rendered: ComponentChild = null;
 
   constructor() {
+    installShallowComponentHooks();
```

Comment:
One comment from #223 was around not including the shallow renderer code unless the user used it. With this changed, I've updated the shallow renderer to now be side-effect free. So only if a use turns on the `useCompatShallowRenderer` flag do these additional hooks and side-effects apply. Do you think it is still valuable to separate out the compat shallow renderer code?

One challenge with separating out the code is that in ESM there is no dynamic synchronous import. And the `createRenderer` method must be synchronous so I can't do something like:
```js
if (useCompatShallowRenderer) {
    const CompatShallowRenderer = await import('./CompatShallowRenderer');
    return new CompatShallowRenderer();
}
```

Another approach if we still want to separate out the code is to do like what `preact` does and have a sub-package in this package called `compat` that re-exports everything from this package and also exposes the compat options (or maybe uses them by default?). With this approach a user would write `import { Adapter } from 'enzyme-adapter-preact-pure/compat';`.

What do you think?

Reply:
> One challenge with separating out the code is that in ESM there is no dynamic synchronous import. And the createRenderer method must be synchronous so I can't do something like:

If the compat shallow renderer is made separately loadable, I would leave it up to the consumer to load it and inject it. So something like:

```js
import { CompatShallowRenderer } from 'enzyme-adapter-preact-pure/compat'
import { configure } from 'enzyme-adapter-preact-pure';

configure({ shallowRenderer: CompatShallowRenderer })
```

The main reason to make the code separately loadable would be to reduce the impact on build times and test bundle sizes. I'll have to do some tests with Hypothesis's apps (where we build a bundle with Rollup and run all tests in an actual browser) to quantify this.

Reply:
Ahh yea loading in the browser. A separate package would be nice for that scenario. Will reorganize the code to move the compat renderer into a new sub package :)